mindmeld/rasaconverter.py

```diff
@@ -0,0 +1,545 @@
+        for entity in entities:
+            entity_path = mindmeld_project_directory + "/entities/" + entity
+            Converter.create_directory(entity_path)
+            with open(entity_path + "/gazetteer.txt", "w") as f:
+                f.close()
+            with open(entity_path + "/mapping.json", "w") as f:
+                # skeleton mapping file that a user must fill in
+                f.write("{\n  \"entities\":[]\n}")
+                f.close()
+
+    @staticmethod
+    def __is_line_intent_definiton(line):
+        return (line[0:10] == "## intent:")
+
+    @staticmethod
+    def __get_intent_from_line(line):
+        return line.split(' ')[1].split(':')[1].rstrip()
+
+    @staticmethod
+    def __create_intent_training_file(intent_directory):
+        Converter.create_directory(intent_directory)
+        with open(intent_directory + "/train.txt", "w") as f:
+            f.close()
+
+    @staticmethod
+    def __does_intent_ex_contain_entity(intent_example):
+        return len(re.findall(r"\[.*\]\(.*\)", intent_example)) > 0
+
+    @staticmethod
+    def __write_intent_with_extinty(intent_f, intent_example):
+        mindmend_intent_example = intent_example
+        for match in re.findall(r"\[.*\]\(.*\)", intent_example):
+            mindmeld_entity = match.replace("[", "{").replace("]", "|") \
+                .replace("(", "").replace(")", "}")
+            mindmend_intent_example = mindmend_intent_example.replace(match, mindmeld_entity)
+            intent_f.write(mindmend_intent_example)
+            # add this to the respective entity gazetteer file as well
+            Rasaconverter.create_entity_files(mindmeld_entity)
+
+    @staticmethod
+    def __remove_comments_from_line(line):
+        start_of_comment = line.find("<!---")
+        end_of_comment = line.find("-->")
+        line_without_comment = line.replace(line[start_of_comment:end_of_comment+3], '')
+        line_without_comment = line_without_comment.rstrip()
+        return line_without_comment
+
+    @staticmethod
+    def __add_example_to_training_file(current_intent_path, line):
+        with open(current_intent_path + "/train.txt", "a") as intent_f:
+            intent_example = line[2:]
+            intent_example = Rasaconverter.__remove_comments_from_line(intent_example) + "\n"
+            if Rasaconverter.__does_intent_ex_contain_entity(intent_example):
+                Rasaconverter.__write_intent_with_extinty(intent_f, intent_example)
+            else:
+                intent_f.write(intent_example)
+
+    def __read_domain_file(self):
+        if os.path.isfile(self.rasa_project_directory + "/domain.yml"):
+            try:
+                with open(self.rasa_project_directory + "/domain.yml", "r") as stream:
+                    domain_data_loaded = yaml.safe_load(stream)
+                    return domain_data_loaded
+            except IOError as e:
+                print("Can not open domain.yml file at %s"
```

Comment:
use logger instead

Reply:
Should I still print(e) or change to logger.error(e)?

Reply:
change to logger.error